src/librustdoc/clean/types.rs

```diff
@@ -2229,10 +2229,25 @@ pub(crate) struct PathSegment {
     pub(crate) args: GenericArgs,
 }
 
+#[derive(Clone, Debug, Default)]
+pub(crate) enum TypedefInnerType {
+    Enum(
+        IndexVec<VariantIdx, Item>,
+        /* has_stripped_entries */ bool,
+        /* is_non_exhaustive */ bool,
+    ),
+    Struct(Option<CtorKind>, Vec<Item>, /* has_stripped_entries */ bool),
```

Comment:
There is also union types. So either you rename this variant into "StructLike" (or something along the line) or you add a `Union` variant.

Reply:
Added a `Union` variant.